### PR TITLE
change all encrypted or decrypted empty strings to null

### PR DESCRIPTION
**Pull request by brewdium**
This I believe is a bit more of a logical change than doing some encryption work around an empty string, it should simplify quite a few things in the future. It should also prove to be a performance benefit in some situations.

### CODE REVIEW

File: test/decryption/decryption.spec.ts [hunk 1]
```diff
@@ -60,6 +60,43 @@ describe('decryption', () => {
     }
   });
 
+  it('returns null if an empty string is passed in as encrypted data to decryptWithKey', async (done) => {
```
Comment (zbarbuto) [inline test/decryption/decryption.spec.ts:63]: This test isn't correct? Serialized should be something like `Aes256Gcm..<...>`

Comment (zbarbuto) [reply test/decryption/decryption.spec.ts:63]: And the assertion `expect(decrypted).toEqual(null)`

File: test/encryption/encryption.spec.ts [hunk 2]
```diff
@@ -90,6 +90,47 @@ describe('Encryption', () => {
...
+      const data = '';
+      const result = await encryptWithKeyUsingArtefacts(
+        key,
+        data,
+        CipherStrategy.AES_GCM,
+        binaryBufferToString(
+          new Uint8Array([13, 120, 218, 57, 166, 132, 154, 162, 228, 63, 63, 143])
+        )
+      );
+      // Known IV and known key should produce the same results
```
Comment (zbarbuto) [inline test/encryption/encryption.spec.ts]: Comment not applicable here.

**Review by vijayshiyani — COMMENTED** [review]
LGTM except for @zbarbuto comments on the test.

**Pull request merged**